src/expressions/xquery/CompTextConstructor.ts

```diff
@@ -0,0 +1,36 @@
+import Expression, { RESULT_ORDERINGS } from '../Expression';
+import Specificity from '../Specificity';
+
+import castToType from '../dataTypes/castToType';
+import createNodeValue from '../dataTypes/createNodeValue';
+import sequenceFactory from '../dataTypes/sequenceFactory';
+
+class CompTextConstructor extends Expression {
+	private _expr: Expression;
+	constructor(expr: Expression | null) {
+		super(expr ? expr.specificity : new Specificity({}), expr ? [expr] : [], {
+			canBeStaticallyEvaluated: false,
+			resultOrder: RESULT_ORDERINGS.UNSORTED
+		});
+
+		this._expr = expr;
+	}
+
+	public evaluate(_dynamicContext, executionParameters) {
```

Comment:
Lets add typing for these. `public evaluate(_dynamicContext: DynamicContext, executionParameters: ExecutionParameters) {`